ecosystem/platform/server/app/controllers/discourse_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+require 'discourse_api'
+
+class DiscourseController < ApplicationController
+
+  before_action :set_post_login_redirect!
+  before_action :authenticate_user!
+  before_action :ensure_confirmed!
+
+  def sso
+    secret = ENV.fetch('DISCOURSE_SECRET', 'MY_SECRET_STRING')
```

Comment:
I don't think there's any advantage to have the fallback, if the env var isn't present then this method should fail.